src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x ? coords.x : 0,
```

Comment:
You can use ?? or || operators here as well
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing
```suggestion
      x: coords.x ?? 0,
```

Reply:
Thank you for an advise. I used || here because the linter test fails with ?? operator. But ?? operator is very usefull I will keep it on mind